USWDS-Site - About: Move release notes section to What’s New

mahoneycm opened the pull request:
# Summary

Moved release notes section into the what’s new page and adds a note about subscribing to releases on GitHub. This PR also removes the two column layout from the What’s New page to accommodate for the new section.

## Related issue

Closes #2136

## Preview link

[What’s new →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-release-notes-update/about/whats-new/)

## Problem statement

The current Release Notes page improperly displays release notes that are pulled in from GitHub.

## Solution

Since this is a low traffic page, we’ve opted to move release notes information to the What’s New page and directly link out to GitHub release notes.

From our [Office Hours meeting notes](https://docs.google.com/document/d/1zveLSehSuFG_mJdSgpPhRm2U8DdZPZ3kToVpy4rqzSA/edit#heading=h.4k11bj3r2nwk) (Google Docs 🔒):

> **Conclusion:** Remove release notes page. Forward the release notes url to “what’s new” page. Add a release notes section to the what’s new page. In that section, add link to release notes in github and info about subscribing to releases.
> 

## Testing and review

1. Visit [What’s New](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-release-notes-update/about/whats-new/) page.
2. Confirm content appropriately informs users of where to find release notes as well as how to subscribe to releases.
3. Check grammar, punctuation, and spelling.
4. Approve of page layout.
5. Inspect [community](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-release-notes-update/about/community/) page.
6. Confirm external link buttons have launch icon and appropriate margins.
7. Confirm “Release notes” page is removed from site navigation.
8. Confirm there are no links to the release notes page.

### Testing checklist

- [ ]  Information on release notes is properly conveyed.
- [ ]  Information on how to subscribe to releases is clear and easy to follow.
- [ ]  What’s New page is well presented.
- [ ]  Prose descendant external links styled as buttons have external link icon.
- [ ]  Standard external links unaffected by this change.
- [ ]  Internal links styled as buttons unaffected by this change.
- [ ]  No grammatical or spelling errors.
- [ ]  Expected redirects are present and working properly.
- [ ]  HTMLProofer runs without error.

amyleadem reviewed (changes requested):
- [x] Confirmed that the releases page was removed from the site
- [ ] Confirmed that redirects have been set up and links to the releases page have been updated
    - There is still a reference to the `whats-new/releases` page on the [implementations page](https://github.com/uswds/uswds-site/blob/main/pages/documentation/implementations.md?plain=1#L34) and [crawl.js](https://github.com/uswds/uswds-site/blob/main/config/crawl.js#L13)
- [ ] Confirmed that the new content makes sense and is free from error
    - I added some notes for possible improvements. I think it would also be good if @finekatie or @sarah-sch could take a look to see if this is the right copy for the page.   I've tagged you both as reviewers in case you are able to take a look!

amyleadem commented on pages/whats-new/overview.md:
```diff
@@ -12,12 +12,20 @@ columns:
-{% capture blog %}
+## Release notes
+
+You can find our release notes — summaries of bug fixes, new features, and other updates introduced in each release on our [GitHub releases page](https://github.com/uswds/uswds/releases).
```
@sarah-sch, @finekatie Can you take a look at this copy in this section and let us know if you have any suggestions?

amyleadem commented on pages/whats-new/overview.md:
```diff
@@ -12,12 +12,20 @@ columns:
+
+You can find our release notes — summaries of bug fixes, new features, and other updates introduced in each release on our [GitHub releases page](https://github.com/uswds/uswds/releases).
+
+You can subscribe to our GitHub releases by visiting the USWDS [GitHub repo](https://github.com/uswds/uswds/), selecting the `watch` button, selecting `custom`, checking the box for `releases`, and selecting apply.
```
@sarah-sch, @finekatie Can you take a look at this copy in this section and let us know if you have any suggestions?

finekatie replied:
Ok, my lack of GitHub is going to show. I can't tell which line 27 above is the change, or if both sections would be part of the page. I saw what was in the preview, and below is my response to that. Apologies if I've confused further!  @sarah-sch , thoughts?
____
You can find summaries of bug fixes, new features, and other updates introduced in each release on our [GitHub releases page](https://github.com/uswds/uswds/releases).

Want to learn about new USWDS releases as they are published? Subscribe now: Visit the USWDS [GitHub repo](https://github.com/uswds/uswds/). Select the watch button, select custom, check the box for releases, and select apply.

mahoneycm replied:
I combined the ideas so that it followed @finekatie's copy recommendation and @amyleadem's idea to rely on GH instructions on subscribing to releases 👍 

## Release notes

You can find summaries of bug fixes, new features, and other updates introduced in each release on our [GitHub releases page](https://github.com/uswds/uswds/releases).

Want to learn about new USWDS releases as they are published? You can sign up to be notified of new USWDS releases by setting up [watch notifications](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#configuring-your-watch-settings-for-an-individual-repository) in the USWDS repository in GitHub. 

--- 

Resolved in [34e5698](https://github.com/uswds/uswds-site/pull/2318/commits/34e56981c3d8e9dd963ba7d5422777097789db8e)

amyleadem replied:
Looks good to me!

mahoneycm reviewed (commented):
Resolved remaining references to `whats-new/releases`, corrected release notes links, and updated copy for new Releases section.

Thanks for the recommendations @amyleadem @finekatie

mahoneycm commented:
## External link button

I realized that since the button is an external link, it should include the external link class and `launch` icon. 

The icon is not displaying currently and I'm working to resolve it now.

Current:

![Screenshot 2023-11-06 at 10 58 30 AM](https://github.com/uswds/uswds-site/assets/25211650/942a43cd-3838-45e9-a033-d831c24def6d)

Expected behavior:
![image](https://github.com/uswds/uswds-site/assets/25211650/06ff8cf2-3878-4fc4-a6fa-021a5b3d899a)

mahoneycm commented:
## External link button cont.

It looks like we're overriding the external link styles for buttons on site:

https://github.com/uswds/uswds-site/blob/b5eabb1f91440ec7a96e7a60521c627b909fbaa5/css/_uswds-theme-custom-styles.scss#L1218-L1224

If we remove `usa-button` from that style definition it displays properly.

![image](https://github.com/uswds/uswds-site/assets/25211650/a956a76e-2098-4fc1-ae4b-b7cc42e623f4)


Can you think of a reason we wouldn't want the external link icon on external links styled as buttons?

@amyleadem @mejiaj

amyleadem commented on _security_updates/2021-10-27-santized-combo-box-content.md:
```diff
@@ -10,7 +10,7 @@ tags:
 
 ## When we made this change
-Fixed in [USWDS 2.12.1]({{ site.baseurl }}/about/releases/#version-uswds-2121).
+Fixed in [USWDS 2.12.1](https://github.com/uswds/uswds/releases/tag/v2.13.1).
```
Resolved in https://github.com/uswds/uswds-site/pull/2318/commits/e000bc28a9d0faeb1da6e42c37b6eaa5276ad3db

amyleadem commented on _security_updates/2021-11-1-automatic-sanitizing-for-all-javascript-components.md:
```diff
@@ -10,7 +10,7 @@ tags:
 
 ## When we made this change
-Fixed in [USWDS 2.12.2]({{ site.baseurl }}/about/releases/#version-uswds-2122).
+Fixed in [USWDS 2.12.2](https://github.com/uswds/uswds/releases/tag/v2.13.2).
```
Resolved in https://github.com/uswds/uswds-site/pull/2318/commits/e000bc28a9d0faeb1da6e42c37b6eaa5276ad3db

amyleadem commented:
@mahoneycm I like the idea of including the external link icon when `usa-button` points to an href outside our domain. I did a quick trial of just removing the `.usa-button` (line 1218) from that style rule and immediately noticed that the github icon in the nav header received an external link icon. I'm wondering if we should either narrow the "display:none" rule to just where it is needed or do an override just for something like `usa-buttons` inside of `usa-prose`.

mahoneycm commented:
## External link buttons styles

Ok I added some styles specific to external links styled as buttons which are descendants of `usa-prose` (Thanks for the idea @amyleadem !)

Additionally, I skimmed through parts of the site for other external links styled as buttons and noticed some visual discrepancies brought on by utility classes. With these additional styles I was able to safely remove the utility classes 👍

I also added the `site-button` class to match other site buttons. 

<details><summary>Screenshots</summary>
<p>

Community page before:

![image](https://github.com/uswds/uswds-site/assets/25211650/895fbfcb-ca3e-4771-a294-dda0123d3121)

Community page after:

![image](https://github.com/uswds/uswds-site/assets/25211650/5f642aea-5d60-4a57-a981-c6b16299b755)


</p>
</details>

amyleadem commented on pages/about/community.md:
```diff
@@ -28,7 +28,7 @@ If you're having trouble, email [{{ site.uswds_email }}](mailto:{{ site.uswds_em
 
 <p>
-  <a href="https://github.com/uswds/uswds" class="usa-button usa-button--outline width-card-lg margin-top-2">USWDS on GitHub</a>
+  <a href="https://github.com/uswds/uswds" class="usa-button site-button usa-button--outline">USWDS on GitHub</a>
```
Note: The `site-button` class changes the font weight. I think it looks better to have the default bolder font on the outline button variant.

![image](https://github.com/uswds/uswds-site/assets/93996430/c41cea85-78fc-4412-8f1a-b22008eb68d9)

amyleadem replied:
Resolved in https://github.com/uswds/uswds-site/pull/2318/commits/024d4d1d26277dc8c23bcef9a6b1b21dd54a978e

amyleadem commented on css/_uswds-theme-custom-styles.scss:
```diff
@@ -1232,6 +1232,16 @@ code[class*="language-plaintext"] {
   }
 }
 
+.usa-prose .usa-button {
+  &#{$external-href} {
+    display: inline-block;
+
+    &::after {
+      display: initial;
+    }
+  }
+}
+
```
After diving a bit deeper into this, I lean away from creating a new rule specific to `usa-prose` and instead lean towards narrowing the `display:none` style rule in [line 1218](https://github.com/uswds/uswds-site/blob/main/css/_uswds-theme-custom-styles.scss#L1218) to something like `nav .usa-button`. 

I think it is a cleaner solution for the code - that way we aren't turning it off everywhere then than turning it on again, and it _might_ cause fewer interactions in the future. Curious what you think!

amyleadem replied:
Resolved in https://github.com/uswds/uswds-site/pull/2318/commits/9aa629e47058dfcb48d1b91b6487c8fefc3ce9ca

amyleadem reviewed (changes requested):
Looking good! Just a couple of small things for your consideration.

amyleadem commented on css/settings/_variables.scss:
```diff
@@ -6,7 +6,10 @@ $asset-path: "../";
 $image-path: "#{$asset-path}img";
 
-$external-href: '[href^="http"]:not([href*="designsystem.digital.gov"])';
+$touchpoints-href: '[href^="https://touchpoints.app.cloud.gov/touchpoints"]';
+$federalist-href: '[href^="https://federalist"]';
+$site-href: '[href*="designsystem.digital.gov"]';
+$external-href: '[href^="http"]:not(#{$site-href}, #{$touchpoints-href}, #{$federalist-href})';
```
These seem to work as expected, but I am curious about using both `href*=` and `href^=` here.

mejiaj replied:
It would be more manageable long term to check for all `https` that have `.gov`.


```suggestion
$gov-href: '[href*=".gov"]';
$external-href: '[href^="https://"]:not(#{$gov-href})';
```

mahoneycm replied:
## Using both `href*=` and `href^=`

I decided to use each of these selectors to be more definitive of what I'm targeting. I believe we could potentially use `*=` for all of them, but the prefix selector seemed to be appropriate in my opinion.

`href*=` - Checks for value anywhere within the `href` value
> Represents elements with an attribute name of attr whose value contains at least one occurrence of value within the string. [^1]

`href^=` - This specifically checks the beginning of the `href` value
> Represents elements with an attribute name of attr whose value is prefixed (preceded) by value. [^2]

[^1]: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_6

[^2]: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_4

## Targetting non `.gov` links

Implementing @mejiaj's suggestion causes a visual regression when linking to `.gov` sites that aren't the design system page

![image](https://github.com/uswds/uswds-site/assets/25211650/b2e7b334-0315-432e-a6b0-5250200e4d69)

The goal with my approach is to add the external link styles to all links that are not:

- The USWDS Site
- The touchpoints survey (visual regression)
- A link to a component/template demo page (visual regression)

All other external links should maintain the external link styles.

We _should_ update the Federalist link to `pages.cloud.gov` so that other links to `federalist` sites still maintain the external link styles. The Federalist link on the implementations page had unintentionally lost its icon.

Updated in [e0ffd5c](https://github.com/uswds/uswds-site/pull/2318/commits/e0ffd5c6a474212fc061c220a1e04fd23e70e77b)

amyleadem reviewed (approved):
This looks good to me! Thanks for your work on this. I had one question below but don't think it needs to be a blocker.

amyleadem commented:
Realized that this PR does not have a changelog. Will work on adding one now.

amyleadem commented:
Update: Added a changelog. Also re-instated the in-page nav on the page now that the content is longer and laid out in a more vertical format.

mejiaj reviewed (changes requested):
Minor comment on external links check. Also found this quirk when clicking on Sidenav `What's new`:

<img width="247" alt="image" src="https://github.com/uswds/uswds-site/assets/3385219/307e8968-f569-4da7-90b9-2dca0f70b149">


**How to reproduce**

1. Visit [Preview →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-release-notes-update/about/whats-new/)
2. In sidenav, confirm two missing sections _Whats new_ and `Empty link`.

sarah-sch commented on pages/whats-new/overview.md:
```diff
@@ -12,12 +12,26 @@ columns:
-{% capture blog %}
+## Release notes
+
+You can find summaries of bug fixes, new features, and other updates introduced in each release on our [GitHub releases page](https://github.com/uswds/uswds/releases).
```
New suggestion for first paragraph, but interested to know if @finekatie has any tweaks: 

Find summaries of each USWDS update on our [GitHub releases page](https://github.com/uswds/uswds/releases). Release notes explain bug fixes, new features, and other changes.

mahoneycm replied:
Updated in [11a0ab7](https://github.com/uswds/uswds-site/pull/2318/commits/11a0ab7f4583c51362114a715cc49f201856b06b)

sarah-sch reviewed (changes requested):
Suggesting updated language for first paragraph under "Release notes."

finekatie commented:
Information on release notes is properly conveyed.
 Information on how to subscribe to releases is clear and easy to follow.
 No grammatical or spelling errors.
 
 I do notice that the first CTA goes to live link. The other two go to federalist links. Is that because the changes/updates are being made to the Release Notes section only?

mahoneycm commented:
@finekatie @sarah-sch Ready for re-review!

And to answer your question Katie

> I do notice that the first CTA goes to live link. The other two go to federalist links. Is that because the changes/updates are being made to the Release Notes section only?

The two Federalist links are due to them being internal links. Since the new CTA is to GitHub, it'll actually take you to the live site 👍

finekatie reviewed (approved):
Charlie answered my question about CTA links, so I have no other input. Ok from here.

sarah-sch reviewed (approved):
Approving changes--looks good.

thisisdano commented on _data/changelogs/about-whats-new.yml:
```diff
@@ -0,0 +1,9 @@
+type: documentation
+changelogURL:
+items:
+  - date: 2023-11-21
```
```suggestion
  - date: 2024-02-26
```
